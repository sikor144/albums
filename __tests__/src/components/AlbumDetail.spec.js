import React from 'react';
import renderer from 'react-test-renderer';

import AlbumDetail from '../../../src/components/AlbumDetail';

describe('AlbumDetail component', () => {
  it('renders component correctly', () => {
    const album = {
      title: 'Some title',
      artist: 'Some artist',
      url: 'Some url',
      thumbnail_image: 'Thumbnail',
    };
    const component = renderer.create(
      <AlbumDetail album={album} />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
