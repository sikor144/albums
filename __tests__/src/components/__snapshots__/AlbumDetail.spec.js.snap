// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDetail component renders component correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "Thumbnail",
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Some title
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Some artist
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 300,
          "width": null,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Buy Now
      </Text>
    </View>
  </View>
</View>
`;
